starknet_py/net/client.py

```diff
@@ -153,7 +151,7 @@ async def wait_for_tx(
                     TransactionStatus.ACCEPTED_ON_L1,
                     TransactionStatus.ACCEPTED_ON_L2,
                 ):
-                    return result.block_number, status
+                    return cast(int, result.block_number), status
```

Comment:
I added an assertion as a sanity check. I don't want to add more code validating this because having a block_number is an invariant there. I think using assertions in cases like this is a perfect fit.

Reply:
I'm not 100% sure about using `assert` in production code, but if that resolves the issue, I think we can keep it.